lib/events/complete.go

```diff
@@ -180,6 +199,7 @@ func (u *UploadCompleter) checkUploads(ctx context.Context) error {
 		}
 		u.log.Debugf("Completed upload for session %v.", upload.SessionID)
 		completed++
+		incompleteSessionUploads.Dec()
```

Comment:
You can probably also decrement this right before the continue on line 191. (That typically happens if another auth server completed this upload after we listed uploads and before we got to completing it ourselves).